util/dvsim/sim_utils.py

```diff
@@ -99,3 +100,116 @@ def vcs_cov_summary_table(buf):
+    for line in reversed(log_text):
+        if "CPU Time" in line:
+            # Find pattern: CPU Time: "XXXX" seconds;.
+            m = re.search(r"(\d+\.\d+)", line)
+            if m:
+                job_runtime = float(m.group(0))
+                return job_runtime
+            else:
+                raise SyntaxError(f"Cannot find job runtime in line: " + line)
+    raise SyntaxError(f"Cannot find job runtime in this file")
+
+
+def xcelium_job_runtime(log_text: List) -> float:
+    """Capture and return the job_runtime from Xcelium simulator.
+
+    Example of Xcelium job_runtime:
+      'TOOL: xrun(64) 21.09-s006: Exiting on Jul 22, 2022 at 13:38:48 PDT (total: 00:00:02)'
+    """
+    # Reverse the log_text because the runtime is usually printed at the end of
+    # the file.
+    for line in reversed(log_text):
+        if "total:" in line and "TOOL" in line:
+            # Find pattern: TOOL .. (total: "XX":"XX":"XX")
+            m = re.search(r"total:.*(\d+):(\d+):(\d+)", line)
+            if m:
+                # Convert time format: HR:MIN:SEC to seconds.
+                val = re.findall(r"\d+", m.group(0))
+                job_runtime = (int(val[0]) * 60 + int(val[1])) * 60 + int(val[2])
+                return job_runtime
+            else:
+                raise SyntaxError(f"Cannot find job runtime in line: " + line)
+    raise SyntaxError(f"Cannot find job runtime in this file")
+
+
+def get_simulated_time(log_text: List, tool: str) -> float:
+    """Capture and return the job simulated_time.
+
+    The log file is passed as an input to the function, in addition to the tool used. The tool
+    returns a number if the simulated time was extracted successfully. The function will raise
+    appropriate exception if the extraction fails.
+    """
+    if tool == 'xcelium':
+        return xcelium_simulated_time(log_text)
+    elif tool == 'vcs':
+        return vcs_simulated_time(log_text)
+    else:
+        raise NotImplementedError(f"{tool} is unsupported for run times extraction.")
+
+
+def xcelium_simulated_time(log_text: List) -> float:
+    """Capture and return the simulated_time from Xcelium simulator.
+
+    Example of Xcelium simulated_time:
+      'UVM_INFO @  51524469 ps: (uvm_report_server.svh:847) [UVM/REPORT/SERVER]'
+    """
+    # Reverse the log_text because Xcelium does not print out the total
+    # simulation time in summary, so we parse the log in reversed order to get
+    # the last simulation timestamp.
+    for line in reversed(log_text):
+        # Extract sim runtime from the last timestamp.
+        if "ps" in line:
+            m = re.search(r"(\d+).*ps", line)
+            if m:
+                simulated_time = "{:.3f}".format(int(m.group(1)) / 1_000_000_000.0)
+                return simulated_time
+            else:
+                raise SyntaxError(f"Cannot find the simulation time in line: " + line)
+    raise SyntaxError(f"Cannot find the simulation time in this log file")
```

Comment:
IMO we can use this as the tool-agnostic way to get the simulated time, since it will show up on all simulators - that way you can reduce 3 (+ n for other simulators) methods into 1.